pkg/fetch/vendir.go

```diff
@@ -351,18 +356,20 @@ func (v *Vendir) configMapBytes(configMapRef vendirconf.DirectoryContentsLocalRe
 // expand this option to other fetch options, we will need to add hostname
 // extraction for those
 func (v *Vendir) shouldSkipTLSVerify(url string) bool {
-	hostname := v.extractImageRefHostname(url)
+	hostname, hostAndPort := v.extractImageRefHostname(url)
 	skip := v.skipTLSConfig.ShouldSkipTLSForDomain(hostname)
```

Comment:
i think we should move all the logic responsible for checking domain/domain+port into ShouldSkipTLSForDomain. probably also would want to use https://pkg.go.dev/net#SplitHostPort there internally.

ShouldSkipTLSForDomain could be rename to ShouldSkipTLSForHostPort (or something along those lines).

Reply:
IMO `SplitHostPort` is less ergonomic than checking for / splitting on `:` ourselves 

https://go.dev/play/p/1Ziio3WSeJ7

Reply:
it covers other cases that strings.Split does not. take a look at the source for SplitHostPort in go docs.

Reply:
these, right: https://cs.opensource.google/go/go/+/refs/tags/go1.17.8:src/net/ipsock.go;l=164

so the two error cases it covers is
const (
		[missingPort](https://cs.opensource.google/go/go/+/refs/tags/go1.17.8:src/net/ipsock.go;bpv=1;bpt=1;l=166?gsn=missingPort&gs=kythe%3A%2F%2Fgolang.org%3Flang%3Dgo%3Fpath%3Dnet%23const%2520%255Bipsock.go%2523166%255D.missingPort)   = "missing port in address"
		[tooManyColons](https://cs.opensource.google/go/go/+/refs/tags/go1.17.8:src/net/ipsock.go;bpv=1;bpt=1;l=167?gsn=tooManyColons&gs=kythe%3A%2F%2Fgolang.org%3Flang%3Dgo%3Fpath%3Dnet%23const%2520%255Bipsock.go%2523167%255D.tooManyColons) = "too many colons in address"
	)
	
	of which, one of them is pretty explicitly not an error case.  and the other one we'd swallow/ignore anyway under the current behavior of the functions (this stack doesn't currently return or surface errors very well, but that's a larger change ... e.g. in testing it i realized that if i pass in `registry.com/myCamelCasedOrg/image:tag` then actually something in gocontainerregistry rejects the URL because it contains capital letters,  but we swallow that error. )

Reply:
just noting here for reference,  i never knew this but i guess `host:port` is called [authority](https://www.rfc-editor.org/rfc/rfc3986#section-3)